infra/identity/src/utils.js

```diff
@@ -216,6 +253,81 @@ async function loadIdentityAttestationsMetadata(addresses, attestations) {
   return metadata
 }
 
+/**
+ * Validates identity data stored in IPFS by checking against JSON schemas.
+ * Throws in case of a validation error.
+ *
+ * @param {Object} ipfsData: JSON parsed identity data stored on IPFS.
+ */
+function validateIdentityIpfsData(ipfsData) {
+  validator('https://schema.originprotocol.com/identity_1.0.0.json', ipfsData)
+  validator(
+    'https://schema.originprotocol.com/profile_2.0.0.json',
+    ipfsData.profile
+  )
+  ipfsData.attestations.forEach(a => {
+    validator('https://schema.originprotocol.com/attestation_1.0.0.json', a)
+  })
+}
+
+/**
+ * Saves an identity in the DB.
+ *
+ * @param {string} owner: Eth address of the user
+ * @param {string} ipfsHash: IPFS hash of the identity blob.
+ * @param {Object} ipfsData: JSON parsed identity blob stored in IPFS.
+ * @param {Object} attestationMetadata: Attestation specific metadata loaded from the DB.
+ * @returns {Promise<{firstName: *, lastName: *, data: {identity: *, ipfsHash: *, ipfsHashHistory: []}, avatarUrl: *, ethAddress: *}>}
+ */
+async function saveIdentity(owner, ipfsHash, ipfsData, attestationMetadata) {
+  // Create an object representing the updated identity.
+  // Notes:
+  //  - By convention, the identity is stored under the owner's address in the DB.
+  //  - We omit a few fields coming from the attestation metadata such as
+  //    twitterProfile and telegramProfilesince. Those are large JSON objects
+  //    that would clutter the identity and they can be retrieved if needed
+  //    from the attestation table.
+  const blacklistedFields = ['twitterProfile', 'telegramProfile']
```

Comment:
Clutter?  Curious what lead to this.

Reply:
The twitterProfile for example is very verbose. It includes lot of data that I don't think we have ever queried via the identity table (for example last tweet from the user). And it makes it hard to look at the data when querying it via postgres cli. Since we were just duplicating the data we already store in the attestation table, for now I decided to not continue doing that.